src/core/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,184 @@
+
+  paginationInputProps?: PaginationInputProps;
+}
+
+// export type PaginationProps = inputProps & InternalPaginationProps;
+export type PaginationProps = InternalPaginationProps;
+
+const baseClassName = 'fi-pagination';
+
+export const paginationClassNames = {
+  styleWrapper: `${baseClassName}_style-wrapper`,
+  content: `${baseClassName}_content`,
+  contentWrapper: `${baseClassName}_contentWrapper`,
+  heading: `${baseClassName}_heading`,
+  textContentWrapper: `${baseClassName}_text-content-wrapper`,
+  icon: `${baseClassName}_icon`,
+  iconWrapper: `${baseClassName}_icon-wrapper`,
+  pageButton: `${baseClassName}_page-button`,
+  stockButton: `${baseClassName}_stock-button`,
+  smallScreen: `${baseClassName}--small-screen`,
+  pageInputWrapper: `${baseClassName}_page-input-wrapper`,
+  buttonsWrapper: `${baseClassName}_buttons-wrapper`,
+  actionElementWrapper: `${baseClassName}_action-element-wrapper`,
+  pageNumbers: `${baseClassName}_page-numbers`,
+};
+
+class BasePagination extends Component<PaginationProps> {
+  private onLeftButtonClick = () => {
+    this.props.onChange(this.props.currentPage - 1);
+  };
+
+  private onRightButtonClick = () => {
+    this.props.onChange(this.props.currentPage + 1);
+  };
+
+  private onNumberInputChange = (page: number) => {
+    this.props.onChange(page);
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      asProp,
+      textFunction,
+      onChange,
+      currentPage,
+      lastPage,
+      ariaPreviousButtonLabel,
+      ariaNextButtonLabel,
+      pageInput,
+      paginationInputProps,
+      smallScreen,
+      ...passProps
+    } = this.props;
+    return (
+      <HtmlNav
+        {...passProps}
+        className={classnames(baseClassName, className, externalClassName, {
+          [paginationClassNames.smallScreen]: !!smallScreen,
+        })}
+      >
+        <HtmlDiv className={paginationClassNames.styleWrapper}>
+          <HtmlDiv className={paginationClassNames.buttonsWrapper}>
+            <Button
+              className={paginationClassNames.stockButton}
+              variant="secondary"
+              onClick={this.onLeftButtonClick}
+              icon="arrowLeft"
+              disabled={currentPage <= 1}
+              aria-label={ariaPreviousButtonLabel}
+            />
+
+            <HtmlSpan
+              aria-live="polite"
+              aria-atomic="true"
+              className={paginationClassNames.pageNumbers}
+            >
+              {textFunction ? textFunction(currentPage, lastPage) : ''}
+            </HtmlSpan>
+
+            <Button
+              className={paginationClassNames.stockButton}
+              variant="secondary"
+              onClick={this.onRightButtonClick}
+              disabled={currentPage >= lastPage}
+              aria-label={ariaNextButtonLabel}
+              icon="arrowRight"
+            />
+          </HtmlDiv>
+
+          {pageInput === true && paginationInputProps && (
```

Comment:
I think the props should probably be handled via conditional props so that the page input props are required when `pageInput===true`. Now it's a bit confusing that the input doesn't render even if one sets `pageInput` to true.

Reply:
This is now conditional prop

Reply:
I think this was discussed at some meeting and this seemed like a common model in other components.